recipes/hdf5/all/conandata.yml

```diff
@@ -1,4 +1,7 @@
 sources:
+  "1.12.1":
+    url: "https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.12/hdf5-1.12.1/src/hdf5-1.12.1.tar.gz"
```

Comment:
Why not take it from GitHub https://github.com/HDFGroup/hdf5/releases/tag/hdf5-1_12_1?